torch/csrc/jit/ir.cpp

```diff
@@ -250,6 +254,220 @@ std::ostream& operator<<(std::ostream & out, const Graph & g) {
+    indent(out, level + 1);
+    prettyPrintValue(out, output_value, constants, aliases);
+    out << " = ";
+    prettyPrintValue(out, if_block->return_node()->inputs()[if_count++], constants, aliases);
+    out << std::endl;
+    if_count++;
+  }
+
+  indent(out, level);
+  out << "else:" << std::endl;
+  prettyPrintBlockVisitor(out, else_block, level + 1, constants, aliases);
+  int else_count = 0;
+  for (const auto* output_value : node->outputs()) {
+    indent(out, level + 1);
+    out << output_value->readableName();
+    out << " = ";
+    prettyPrintValue(out, else_block->return_node()->inputs()[else_count], constants, aliases);
+    out << std::endl;
+    else_count++;
+  }
+
+  return out;
+}
+
+std::ostream& prettyPrintLoop(
+  std::ostream & out,
+  const Node* node,
+  int level,
+  constant_list& constants,
+  alias_list& aliases
+) {
+  out << "while ";
+  prettyPrintValue(out, node->inputs()[1], constants, aliases);
+
+  out << std::endl;
+  auto body_block = node->blocks()[0];
+
+  // Offset to skip loop trip count
+  int count = 1;
+  for (const auto* block_param : body_block->param_node()->outputs())  {
+    // add names to map
+    aliases[block_param->uniqueName()] = node->inputs()[count]->readableName();
+    count++;
+  }
+
+  // Add alias for loop condition
+  aliases[body_block->return_node()->inputs()[0]->readableName()] = node->inputs()[1]->uniqueName();
+
+  // for (auto block_input : body_block->param_node())
+  prettyPrintBlockVisitor(out, body_block, level + 1, constants, aliases);
+
+  return out;
+}
+
+// Handles iterating over nodes/subblocks
+std::ostream& prettyPrintNodeVisitor(
+  std::ostream & out,
+  const Node* node,
+  int level,
+  constant_list& constants,
+  alias_list& aliases
+) {
+  if (node->kind() == prim::Constant) {
+    // Add constant to current scope map
+    constants[node->outputs()[0]->uniqueName()] = std::make_pair(
+      node,
+      node->attributeNames()[0]
+    );
+    return out;
+  }
+
+  if (node->kind() == prim::Return) {
+    // Handled elsewhere, do nothing
+    return out;
+  }
+
+  indent(out, level);
+
+  // if there are subblocks on this node, visit them
+  switch (node->kind()) {
+  case prim::Loop:
+    prettyPrintLoop(out, node, level, constants, aliases);
+    break;
+  case prim::If:
+    prettyPrintIf(out, node, level, constants, aliases);
+    break;
+  default:
+    // Print outputs
+    if (node->outputs().size() > 0) {
+      auto delim = "";
+      for (const auto* output_value : node->outputs()) {
+        out << delim;
+        prettyPrintValue(out, output_value, constants, aliases);
+        delim = ", ";
+      }
+      out << " = ";
+    }
+
+    out << node->kind().toQualString();
+
+    // Print instruction parameters
+    prettyPrintInputs(out, node, constants, aliases);
+
+    out << std::endl;
+  }
+
+  return out;
+}
+
+std::ostream& Graph::prettyPrint(std::ostream & out) {
+  constant_list empty_constants;
+  alias_list empty_aliases;
```

Comment:
Can we have a `Note [Pretty-printing aliases]` explaining what the aliases do?

Reply:
Also, you're passing `empty_aliases` in by mutable reference... so it's not going to be empty after your function is done with it is it?